Sources/NIOTestUtils/ByteToMessageDecoderVerifier.swift

```diff
@@ -0,0 +1,211 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+public enum ByteToMessageDecoderVerifier {
+    /// - seealso: verifyDecoder(inputOutputPairs:decoderFactory:)
+    ///
+    /// Verify `ByteToMessageDecoder`s with `String` inputs
+    public static func verifyDecoder<Decoder: ByteToMessageDecoder>(stringInputOutputPairs: [(String, [Decoder.InboundOut])],
+                                                                    decoderFactory: @escaping () -> Decoder) throws where Decoder.InboundOut: Equatable {
+        let alloc = ByteBufferAllocator()
+        let ioPairs = stringInputOutputPairs.map { (ioPair: (String, [Decoder.InboundOut])) -> (ByteBuffer, [Decoder.InboundOut]) in
+            var buffer = alloc.buffer(capacity: ioPair.0.utf8.count)
+            buffer.writeString(ioPair.0)
+            return (buffer, ioPair.1)
+        }
+        return try ByteToMessageDecoderVerifier.verifyDecoder(inputOutputPairs: ioPairs, decoderFactory: decoderFactory)
+    }
+
+    /// Verifies a `ByteToMessageDecoder` by performing a number of tests.
+    ///
+    /// This method is mostly useful in unit tests for `ByteToMessageDecoder`s. It feeds the inputs from
+    /// `inputOutputPairs` into the decoder in various ways and expects the decoder to produce the outputs from
+    /// `inputOutputPairs`.
+    ///
+    /// The verification is performing various tests, for example:
```

Comment:
Nit: "The verification performs various tests, for example:"

Reply:
done